docs/maui/views/popup/popup-result.md

```diff
@@ -0,0 +1,160 @@
+
+## Building a Popup
+
+A `Popup<T>` can be created in XAML or C# as follows:
+
+### XAML
+
+The following section covers how to create a `Popup<T>` using XAML.
+
+#### Defining your Popup
+
+The easiest way to create a `Popup<T>` is to add a new `.NET MAUI ContentView (XAML)` to your project, this will create 2 files; a _*.xaml_ file and a _*.xaml.cs_ file. The contents of each file can be replaced with the following:
+
+##### .xaml
+
+```xaml
+<toolkit:Popup
+    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+    x:TypeArguments="system:Boolean"
+    x:Class="MyProject.ReturnResultPopup">
+
+    <VerticalStackLayout>
+        <Label Text="This is a very important message! Do you agree?" />
+
+        <Button Text="Yes" 
+                Clicked="OnYesButtonClicked" />
+
+        <Button Text="No"
+                Clicked="OnNoButtonClicked" />
+    
+    </VerticalStackLayout>
+    
+</toolkit:Popup>
+```
+
+##### .xaml.cs
+
+```csharp
+public partial class ReturnResultPopup : ContentView
+{
+    public ReturnResultPopup()
+    {
+        InitializeComponent();
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+}
+```
+
+> [!IMPORTANT]
+> If the code behind file is not created along with the call to `InitializeComponent` then an exception will be thrown when trying to display your `Popup`.
+
+### C#
+
+The following section covers how to create a `Popup` using C#.
+
+```csharp
+using CommunityToolkit.Maui.Views;
+
+namespace MyProject;
+
+public class ReturnResultPopup : Popup<bool>
+{
+    public ReturnResultPopup()
+    {
+        var yesButton = new Button { Text = "Yes" };
+        yesButton.Clicked += OnYesButtonClicked;
+
+        var noButton = new Button { Text = "No" };
+        noButton.Clicked += OnNoButtonClicked;
+
+        Content = new VerticalStackLayout
+        {
+            Children = 
+            {
+                new Label { Text = "This is a very important message! Do you agree?" },
+                
+                yesButton,
+                noButton
+            }
+        };
+    }
+
+    async void OnYesButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(true);
+    }
+    
+    async void OnNoButtonClicked(object? sender, EventArgs e)
+    {
+        await CloseAsync(false);
+    }
+```
+
+The `Close` method allows for an `object` value to be supplied, this will be the resulting return value. 
```

Comment:
I would change `object` for something like `desired type`, because we use generic there and saying that the method allows a `object` could cause confusion, on users thinking that the value will be `object` and not the type that they specify in the Popup's definition. In other words, with

> The `Close` method allows for an `object`
 
I imagine that Close method has this signature
`void Close(object result)`

instead of 
`void Close (T result)`